File: guides/java/kora-java-guide-openapi-http-server-advanced-app/src/test/java/ru/tinkoff/kora/guide/openapi/httpserver/advanced/OpenApiHttpServerAdvancedAppTest.java
package ru.tinkoff.kora.guide.openapi.httpserver.advanced;

import org.junit.jupiter.api.Test;
import ru.tinkoff.kora.guide.openapi.httpserver.data.api.DataApiController;
import ru.tinkoff.kora.guide.openapi.httpserver.data.api.DataApiDelegate;
import ru.tinkoff.kora.guide.openapi.httpserver.data.api.DataApiResponses;
import ru.tinkoff.kora.test.extension.junit5.KoraAppTest;
import ru.tinkoff.kora.test.extension.junit5.TestComponent;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertInstanceOf;

@KoraAppTest(Application.class)
class OpenApiHttpServerAdvancedAppTest {

    @TestComponent
    private DataApiDelegate dataApiDelegate;

    @Test
    void dataFormFlowWorksThroughGeneratedDelegate() throws Exception {
        var response = this.dataApiDelegate.processForm(new DataApiController.ProcessFormFormParam("Ivan"));
        var form200 = assertInstanceOf(DataApiResponses.ProcessFormApiResponse.ProcessForm200ApiResponse.class, response);
        assertEquals("Hello World, Ivan", form200.content());
    }

    @Test
    void dataMappingByCodeReturnsPayloadFor200() throws Exception {
        var response = this.dataApiDelegate.mappingByCode(200);
        var mapping200 = assertInstanceOf(DataApiResponses.MappingByCodeApiResponse.MappingByCode200ApiResponse.class, response);
        assertEquals("Hello from response mapper", mapping200.content().message());
    }
}

File: guides/kotlin/kora-kotlin-guide-openapi-http-server-advanced-app/src/test/kotlin/ru/tinkoff/kora/guide/openapi/httpserver/advanced/OpenApiHttpServerAdvancedAppTest.kt
package ru.tinkoff.kora.guide.openapi.httpserver.advanced

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertInstanceOf
import org.junit.jupiter.api.Test
import ru.tinkoff.kora.guide.openapi.httpserver.data.api.DataApiController
import ru.tinkoff.kora.guide.openapi.httpserver.data.api.DataApiDelegate
import ru.tinkoff.kora.guide.openapi.httpserver.data.api.DataApiResponses
import ru.tinkoff.kora.test.extension.junit5.KoraAppTest
import ru.tinkoff.kora.test.extension.junit5.TestComponent

@KoraAppTest(Application::class)
class OpenApiHttpServerAdvancedAppTest {

    @TestComponent
    lateinit var dataApiDelegate: DataApiDelegate

    @Test
    fun dataFormFlowWorksThroughGeneratedDelegate() {
        val response = dataApiDelegate.processForm(DataApiController.ProcessFormFormParam("Ivan"))
        val form200 =
            assertInstanceOf(DataApiResponses.ProcessFormApiResponse.ProcessForm200ApiResponse::class.java, response)
        assertEquals("Hello World, Ivan", form200.content)
    }

    @Test
    fun dataMappingByCodeReturnsPayloadFor200() {
        val response = dataApiDelegate.mappingByCode(200)
        val mapping200 = assertInstanceOf(
            DataApiResponses.MappingByCodeApiResponse.MappingByCode200ApiResponse::class.java,
            response
        )
        assertEquals("Hello from response mapper", mapping200.content.message)
    }
}
